src/live_restore/live_restore_fs.c

```diff
@@ -1178,68 +1182,104 @@ __live_restore_fs_open_in_destination(WT_LIVE_RESTORE_FS *lr_fs, WT_SESSION_IMPL
+__live_restore_setup_lr_fh_directory(WT_SESSION_IMPL *session, WT_LIVE_RESTORE_FS *lr_fs,
+  const char *name, uint32_t flags, WT_LIVE_RESTORE_FILE_HANDLE *lr_fhp)
 {
     WT_DECL_RET;
-    WT_LIVE_RESTORE_FILE_HANDLE *lr_fh;
-    WT_LIVE_RESTORE_FS *lr_fs;
-    WT_LIVE_RESTORE_FS_LAYER_TYPE which;
-    WT_SESSION_IMPL *session;
-    bool dest_exist, source_exist, have_tombstone, readonly;
+    char *dest_folder_path = NULL;
+    bool dest_exist = false, source_exist = false;
 
-    session = (WT_SESSION_IMPL *)wt_session;
-    lr_fs = (WT_LIVE_RESTORE_FS *)fs;
+    WT_RET_NOTFOUND_OK(
+      __live_restore_fs_has_file(lr_fs, &lr_fs->destination, session, name, &dest_exist));
+    WT_RET_NOTFOUND_OK(
+      __live_restore_fs_has_file(lr_fs, &lr_fs->source, session, name, &source_exist));
 
-    dest_exist = source_exist = false;
-    lr_fh = NULL;
-    have_tombstone = false;
-    WT_UNUSED(have_tombstone);
-    readonly = LF_ISSET(WT_FS_OPEN_READONLY);
-    WT_UNUSED(readonly);
-    WT_UNUSED(which);
+    if (!dest_exist && !source_exist && !LF_ISSET(WT_FS_OPEN_CREATE))
+        WT_RET_MSG(session, ENOENT, "Directory %s does not exist in source or destination", name);
 
-    /* FIXME-WT-13808 Handle WT_FS_OPEN_FILE_TYPE_DIRECTORY */
+    if (!dest_exist) {
+        /* The directory doesn't exist in the destination yet. We need to create it in all cases. */
 
-    /* Set up the file handle. */
-    WT_ERR(__wt_calloc_one(session, &lr_fh));
-    WT_ERR(__wt_strdup(session, name, &lr_fh->iface.name));
-    lr_fh->iface.file_system = fs;
-    lr_fh->file_type = file_type;
+        WT_RET(__live_restore_create_file_path(
+          session, &lr_fs->destination, (char *)name, &dest_folder_path));
 
-    WT_ERR(__wt_rwlock_init(session, &lr_fh->ext_lock));
+        /*
+         * We can't create directories with a WT_FS_OPEN_CREATE call. Instead we do it manually.
+         *
+         * FIXME-WT-13971 Defaulting to permissions 0755. If the folder exists in the source should
+         * we copy the permissions from the source?
+         */
+        mkdir(dest_folder_path, 0755);
 
-    /* FIXME-WT-13823 Handle the exclusive flag and other flags */
+        WT_FILE_HANDLE *fh;
+        WT_ERR(lr_fs->os_file_system->fs_open_file(lr_fs->os_file_system, (WT_SESSION *)session,
+          dest_folder_path, lr_fhp->file_type, flags, &fh));
+
+        lr_fhp->destination.fh = fh;
+    }
+
+    /* There's no need for a hole list. The directory has already been fully copied */
+    lr_fhp->destination.hole_list_head = NULL;
+    lr_fhp->destination.back_pointer = lr_fs;
+    lr_fhp->destination.complete = true;
+
+err:
+    if (dest_folder_path != NULL)
+        __wt_free(session, dest_folder_path);
+    return (ret);
+}
+
+/*
+ * __live_restore_setup_lr_fh_file --
+ *     Populate a live restore file handle for a normal (non-directory) file.
+ */
+static int
+__live_restore_setup_lr_fh_file(WT_SESSION_IMPL *session, WT_LIVE_RESTORE_FS *lr_fs,
```

Comment:
I have noticed us being inconsistent with `WT_SESSION_IMPL` vs `WT_SESSION` args, in this case the arg originally was a WT_SESSION only to get cast to and back again. Which is fine, we're also inconsistent with `lr_fh`s coming before WT_SESSION_IMPLs in the arg list. I think I'll make a cleanup ticket but there's no action for you here.

Reply:
Created: https://jira.mongodb.org/browse/WT-13974